class.c

```diff
@@ -636,184 +636,264 @@ Perl_class_apply_attributes(pTHX_ HV *stash, OP *attrlist)
+        class_cleanup_definition(stash);
+        return;
+    }
 
-        resume_compcv_final(aux->xhv_class_suspended_initfields_compcv);
+    struct xpvhv_aux *aux = HvAUX(stash);
 
-        /* Some OP_INITFIELD ops will need to populate the pad with their
-         * result because later ops will rely on it. There's no need to do
-         * this for every op though. Store a mapping to work out which ones
-         * we'll need.
-         */
-        PADNAMELIST *pnl = PadlistNAMES(CvPADLIST(PL_compcv));
-        HV *fieldix_to_padix = newHV();
-        SAVEFREESV((SV *)fieldix_to_padix);
-
-        /* padix 0 == @_; padix 1 == $self. Start at 2 */
-        for(PADOFFSET padix = 2; padix <= PadnamelistMAX(pnl); padix++) {
-            PADNAME *pn = PadnamelistARRAY(pnl)[padix];
-            if(!pn || !PadnameIsFIELD(pn))
-                continue;
-
-            U32 fieldix = PadnameFIELDINFO(pn)->fieldix;
-            (void)hv_store_ent(fieldix_to_padix, sv_2mortal(newSVuv(fieldix)), newSVuv(padix), 0);
-        }
+    /* generate initfields CV */
+    I32 floor_ix = PL_savestack_ix;
+    SAVEI32(PL_subline);
+    save_item(PL_subname);
 
-        OP *ops = NULL;
+    resume_compcv_final(aux->xhv_class_suspended_initfields_compcv);
 
-        ops = op_append_list(OP_LINESEQ, ops,
-                newUNOP_AUX(OP_METHSTART, OPpINITFIELDS << 8, NULL, NULL));
+    /* Some OP_INITFIELD ops will need to populate the pad with their
+     * result because later ops will rely on it. There's no need to do
+     * this for every op though. Store a mapping to work out which ones
+     * we'll need.
+     */
+    PADNAMELIST *pnl = PadlistNAMES(CvPADLIST(PL_compcv));
+    HV *fieldix_to_padix = newHV();
+    SAVEFREESV((SV *)fieldix_to_padix);
 
-        if(aux->xhv_class_superclass) {
-            HV *superstash = aux->xhv_class_superclass;
-            assert(HvSTASH_IS_CLASS(superstash));
-            struct xpvhv_aux *superaux = HvAUX(superstash);
+    /* padix 0 == @_; padix 1 == $self. Start at 2 */
+    for(PADOFFSET padix = 2; padix <= PadnamelistMAX(pnl); padix++) {
+        PADNAME *pn = PadnamelistARRAY(pnl)[padix];
+        if(!pn || !PadnameIsFIELD(pn))
+            continue;
 
-            /* Build an OP_ENTERSUB */
-            OP *o = newLISTOPn(OP_ENTERSUB, OPf_WANT_VOID|OPf_STACKED,
-                newPADxVOP(OP_PADSV, 0, PADIX_SELF),
-                newPADxVOP(OP_PADHV, OPf_REF, PADIX_PARAMS),
-                /* TODO: This won't work at all well under `use threads` because
-                 * it embeds the CV * to the superclass initfields CV right into
-                 * the optree. Maybe we'll have to pop it in the pad or something
-                 */
-                newSVOP(OP_CONST, 0, (SV *)superaux->xhv_class_initfields_cv),
-                NULL);
+        U32 fieldix = PadnameFIELDINFO(pn)->fieldix;
+        (void)hv_store_ent(fieldix_to_padix, sv_2mortal(newSVuv(fieldix)), newSVuv(padix), 0);
```

Comment:
is there a better way of doing this other than a trip through ```SvPV(3args)```'s ```S_uiv_2buf()```, or worse going to libc sprintf with dozens of lock aq/releases, and maybe some kernel calls to the FS. I would just have 4 byte / 8 byte UVs stored straight as raw unprintable binary with SvPOK_on(). Hash algo/perl just sees char strings, vs turning them into longer less dense ascii bytes.

Reply:
This code was only re-indented